PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/storage/component/BaseStorageAccountCreationDialog.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.intellij.storage.component;
+
+import com.intellij.openapi.project.Project;
+import com.microsoft.azure.toolkit.intellij.appservice.region.RegionComboBox;
+import com.microsoft.azure.toolkit.intellij.appservice.resourcegroup.ResourceGroupComboBox;
+import com.microsoft.azure.toolkit.intellij.appservice.subscription.SubscriptionComboBox;
+import com.microsoft.azure.toolkit.intellij.common.AzureDialog;
+import com.microsoft.azure.toolkit.lib.common.form.AzureForm;
+import com.microsoft.azure.toolkit.lib.common.form.AzureFormInput;
+import com.microsoft.azure.toolkit.lib.common.model.Subscription;
+import com.microsoft.azure.toolkit.lib.storage.model.Kind;
+import com.microsoft.azure.toolkit.lib.storage.model.Performance;
+import com.microsoft.azure.toolkit.lib.storage.model.StorageAccountConfig;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ItemEvent;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+public class BaseStorageAccountCreationDialog extends AzureDialog<StorageAccountConfig> implements AzureForm<StorageAccountConfig> {
+
+    private static final String DIALOG_TITLE = "Create Storage Account";
+
+    private JPanel rootPanel;
+    private SubscriptionComboBox subscriptionComboBox;
+    private ResourceGroupComboBox resourceGroupComboBox;
+    private AccountNameTextField accountNameTextField;
+    private RegionComboBox regionComboBox;
+    protected PerformanceComboBox performanceComboBox;
```

Comment:
why some fields are private and some are protected?

Reply:
protected fields will be used in its sub classes.